### PR TITLE
feat: bring back cypress command createDescription through UI

### PR DESCRIPTION
**Pull request by luka-nextcloud**
### 📝 Summary

* Resolves: #4937

#### 🖼️ Screenshots

🏚️ Before | 🏡 After
---|---
B | A

### 🚧 TODO

- [ ] ...

### 🏁 Checklist

- [ ] Code is properly formatted (`npm run lint` / `npm run stylelint` / `composer run cs:check`)
- [ ] [Sign-off message](https://probot.github.io/apps/dco/) is added to all commits
- [ ] [Tests](https://github.com/nextcloud/text#-testing-the-app) (unit, integration and/or end-to-end) passing and the changes are covered with tests
- [ ] Documentation ([README](https://github.com/nextcloud/text/blob/main/README.md) or [documentation](https://github.com/nextcloud/documentation/blob/master/admin_manual/configuration_server/text_configuration.rst#L2)) has been updated or is not required

### CODE REVIEW

File: src/views/RichWorkspace.vue [hunk 1]
```diff
@@ -144,9 +143,10 @@ export default {
 			})
 		},
 		getFileInfo(autofocus) {
-			if (!this.hasRichWorkspace) {
```
Comment (juliushaertl) [inline src/views/RichWorkspace.vue:147]: Can you elaborate a bit why this change is needed? as far as I see this would no longer check on each folder change on if the folder has a rich workspace

File: src/helpers/files.js [hunk 2]
```diff
@@ -177,19 +177,21 @@ const addMenuRichWorkspace = () => {
...
 			})
-			context._children.push(file.fileid)
 
 			showSuccess(t('text', 'Created "{name}"', { name: descriptionFile }))
+
 			emit('files:node:created', file)
-			emit('Text::showRichWorkspace', { autofocus: true })
+			setTimeout(() => {
+				emit('Text::showRichWorkspace', { autofocus: true })
+			}, 200)
```
Comment (juliushaertl) [inline src/helpers/files.js:194]: Can you tell why the previous one didn't work?

File: src/init.js [hunk 3]
```diff
@@ -12,5 +12,6 @@ registerDavProperty('nc:rich-workspace', { nc: 'http://nextcloud.org/ns' })
 registerDavProperty('nc:rich-workspace-file', { nc: 'http://nextcloud.org/ns' })
 
 if (workspaceAvailable) {
+	addMenuRichWorkspace()
```
Comment (juliushaertl) [inline src/init.js:15]: Good idea to move that over here 👍

**Review by juliushaertl — CHANGES_REQUESTED** [review]
Would be great to get a bit of further insights on the changes (see inline)

File: src/views/RichWorkspace.vue (hunk 1 above)
Comment (luka-nextcloud) [reply src/views/RichWorkspace.vue:147]: As I checked,
- when Readme.md file was created, it did not trigger updated function (at https://github.com/nextcloud/text/blob/main/src/helpers/files.js#L232) to update property hasRichWorkspace's value to true.
- So, the RichWorkspace component was not be shown even though Text::showRichWorkspace was triggered since the property hasRichWorkspace was still false (until page refreshed)

File: src/helpers/files.js (hunk 2 above)
Comment (luka-nextcloud) [reply src/helpers/files.js:194]: When user directly accessed an empty folder, the render() (at https://github.com/nextcloud/text/blob/main/src/helpers/files.js#L207) did not run. So, RichWorkspace component was not created.
Then, triggering `Text::showRichWorkspace` right after creating Readme.md did not work.
The RichWorkspace component was created only after any file or folder was created.
So, we have to wait some time until RichWorkspace component is created.

Comment (juliushaertl) [reply src/helpers/files.js]: Ah i see, that seems tricky. Let's get this in then for now. We might want to have the files app adjusted so that the header component itself is also rendered for empty file lists mid term. I only pushed a commit to use the file created event instead which seemed more clean.

**Review by juliushaertl — APPROVED** [review]

**Review by mejo- — APPROVED** [review]
Code changes look good to me :ok_hand:

**Comment by juliushaertl** [comment]
@luka-nextcloud just FYI I found a reasonable workaround for updating the view without the timestamp (see last commit)
